pkg/cmd/adm/install_operator.go

```diff
@@ -0,0 +1,239 @@
+	if err := ctx.KubeClient.Create(ctx, subscription); err != nil {
+		return err
+	}
+	if err := waitUntilInstallPlanIsComplete(ctx, ctx.KubeClient, namespace, timeout); err != nil {
+		return err
+	}
+	ctx.Println(fmt.Sprintf("InstallPlan for the %s operator has been completed", operator))
+	ctx.Println("")
+	ctx.Println(fmt.Sprintf("The %s operator has been successfully installed in the %s namespace", operator, namespace))
+	return nil
+}
+
+func operatorResourceName(operator string) string {
+	return fmt.Sprintf("%s-operator", operator)
+}
+
+func newCatalogSource(name types.NamespacedName, operator string) *olmv1alpha1.CatalogSource {
+	return &olmv1alpha1.CatalogSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1alpha1.CatalogSourceSpec{
+			SourceType:  olmv1alpha1.SourceTypeGrpc,
+			Image:       fmt.Sprintf("quay.io/codeready-toolchain/%s-operator-index:latest", operator),
+			DisplayName: fmt.Sprintf("KubeSaw %s Operator", operator),
+			Publisher:   "Red Hat",
+			UpdateStrategy: &olmv1alpha1.UpdateStrategy{
+				RegistryPoll: &olmv1alpha1.RegistryPoll{
+					Interval: &metav1.Duration{
+						Duration: 5 * time.Minute,
+					},
+				},
+			},
+		},
+	}
+}
+
+func newOperatorGroup(name types.NamespacedName) *olmv1.OperatorGroup {
+	return &olmv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				name.Namespace,
+			},
+		},
+	}
+}
+
+func newSubscription(name types.NamespacedName, operatorName, catalogSourceName string) *olmv1alpha1.Subscription {
+	return &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: &olmv1alpha1.SubscriptionSpec{
+			Channel:                "staging",
+			InstallPlanApproval:    olmv1alpha1.ApprovalAutomatic,
+			Package:                operatorName,
+			CatalogSource:          catalogSourceName,
+			CatalogSourceNamespace: name.Namespace,
+		},
+	}
+}
+
+func waitUntilCatalogSourceIsReady(ctx *clicontext.TerminalContext, catalogSourceKey runtimeclient.ObjectKey, waitForReadyTimeout time.Duration) error {
+	cs := &olmv1alpha1.CatalogSource{}
+	if err := wait.PollImmediate(2*time.Second, waitForReadyTimeout, func() (bool, error) {
+		ctx.Printlnf("waiting for CatalogSource %s to become ready", catalogSourceKey)
+		if err := ctx.KubeClient.Get(ctx, catalogSourceKey, cs); err != nil {
+			return false, err
+		}
+
+		return cs.Status.GRPCConnectionState != nil && cs.Status.GRPCConnectionState.LastObservedState == "READY", nil
+	}); err != nil {
+		csString, _ := json.Marshal(cs)
+		return errs.Wrapf(err, "failed waiting for catalog source to be ready.\n CatalogSrouce found: %v \n\t", string(csString))
```

Comment:
we don't need to wrap it (and if we need to, then let's use the fmt.Errorf with the `%w` placehodler)
```suggestion
		return fmt.Errorf("failed waiting for catalog source to be ready.\n CatalogSrouce found: %v \n\t", string(csString))
```